interface/src/app/gene/search/search.js

```diff
@@ -264,18 +264,21 @@ angular.module('adage.gene.search', [
           for (var gene in $scope.selectedGenes) {
```

Comment:
I should clarify my concern about `hasOwnProperty()`. I do not think it has any inherent performance trouble, in and of itself. The issue I was trying to highlight is that it would add unnecessary steps in any loop that could have performance implications, especially if it is an unnecessary step.

JavaScript has been around a long time and it is now used far more extensively than originally planned. Many of the warnings about its use stem from things you need to be careful about when interacting with the DOM and external libraries, and I count this *for-in without filter* rule among them.

In this context, `$scope.selectedGenes` is created by our own code in `search.js` (line 105):

```
var selectedGenes = {};
```

In a case like this, there is nothing to be concerned about. The use of `{}` for creating an empty Object is the preferred way to do this (because it is more readable and compact than `var selectedGenes = new Object;`. While it is technically true that some library we are using could muck around with the prototype on `Object`, that practice is deeply frowned upon these days and if we find any library we use doing that we should look for an alternative.

With all that said, if the loop is well bounded and not performance-critical (i.e. you're not looping over 600 `Activity` records per sample as I am on the heatmap) then there is no performance concern adding the `hasOwnProperty` filter. It will burn a few extra cycles that do nothing, but for a loop with only a few hundred iterations nobody will notice. (Unless that loop is then put inside another loop for any reason.)